lua/nvim-treesitter/configs.lua

```diff
@@ -295,24 +299,17 @@ end
 -- Setup call for users to override module configurations.
 -- @param user_data module overrides
 function M.setup(user_data)
-  for name, data in pairs(user_data) do
-    if name == 'ensure_installed' then
-      config.ensure_installed = data
-      require'nvim-treesitter.install'.ensure_installed(data)
-    else
-      config.modules[name] = vim.tbl_deep_extend('force', config.modules[name] or {}, data)
-
-      recurse_modules(function(_, _, new_path)
-        if data.enable then
-          enable_all(new_path)
-        end
-
-        for _, lang in ipairs(data.disable or {}) do
-          disable_mod_conf_autocmd(new_path, lang)
-        end
-      end, config.modules)
+  config.modules = vim.tbl_deep_extend('force', config.modules, user_data)
```

Comment:
Why do we have modules here? This causes our table structure to be slightly different from the one the user provides us and it's only for one boolean value.

Reply:
We need that because of the use of the recurse module below

Reply:
Recurse modules does not care about a boolean because it will do a `is_module` check.

Reply:
Then you can even remove the boolean afterwards to save us some time